Assets/Global/Scripts/WaveSpawnArea.cs

```diff
@@ -0,0 +1,63 @@
+using UnityEngine;
+using System.Collections.Generic;
+using Random = UnityEngine.Random;
+
+public class WaveSpawnArea : MonoBehaviour
+{
+    [Header("Spawner Settings")]
+    [Tooltip("Prefab of the enemy to spawn")]
+    [SerializeField] public GameObject enemyPrefab;
+
+    [SerializeField] public int maxSpawnedEnemies;
+
+    [Header("Spawn Area Settings")]
+    [SerializeField] public Transform spawnArea;
+
+    private List<GameObject> activeEnemies = new List<GameObject>();
+
+    public bool center = false;
+
+    public bool waveDone = false;
+
+    private void Update()
+    {
+        if(activeEnemies.Count == maxSpawnedEnemies && activeEnemies.TrueForAll(enemy => enemy == null) &&waveDone)
+        {
+            waveDone = false;
+            GlobalReference.AttemptInvoke(Events.WAVE_DONE);
+            Debug.Log("All enemies are dead");
+        }
+    }
+
+
+    public void SpawnEnemy()
```

Comment:
What happens when the vector is 0? does it retry somewhere?

Reply:
removed the if statement, because it will never become 0,0,0 or your spawner is there with a very small chance of going there